lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -2538,40 +2536,58 @@ impl InstanceDef {
         self
     }
 
-    fn emit_event_impl(
-        event: &event::SomeEvent,
-        parent: Option<Instance>,
-        capturing_event_fan: &frp::Fan,
-        bubbling_event_fan: &frp::Fan,
-    ) {
-        let rev_parent_chain = parent.map(|p| p.rev_parent_chain()).unwrap_or_default();
-        if event.captures.get() {
-            for object in &rev_parent_chain {
-                if !event.is_cancelled() {
-                    event.set_current_target(Some(object));
-                    object.event.capturing_fan.emit(&event.data);
-                } else {
-                    break;
-                }
+    fn emit_event_impl(&self, event: &event::SomeEvent) {
```

Comment:
This needs proper docs now. The code inside is nicely documented, but this becomes really complex and I'd be very thankful for some docs explaining everything it does in the way it does.